src/wutil.h

```diff
@@ -145,10 +145,15 @@ struct file_id_t {
     dev_t device{static_cast<dev_t>(-1LL)};
     ino_t inode{static_cast<ino_t>(-1LL)};
     uint64_t size{static_cast<uint64_t>(-1LL)};
```

Comment:
please don't @ me in commit messages because then I get notifications when any fork pushes that commit

> I'm squashing down the commit history here a bit.

yeah I think that's a good idea, it will make the change easier to review.
Especially since some changes cancel each outer out

Reply:
Oh dang I hate it when that happens, sorry about that.

The twists and turns of how this PR was finally implemented are squashed away now.